lib/libc/common/CMakeLists.txt

```diff
@@ -6,3 +6,6 @@ zephyr_library_sources_ifdef(CONFIG_COMMON_LIBC_ABORT source/stdlib/abort.c)
 zephyr_library_sources_ifdef(CONFIG_COMMON_LIBC_TIME source/time/time.c)
 zephyr_library_sources_ifdef(CONFIG_COMMON_LIBC_MALLOC source/stdlib/malloc.c)
 zephyr_library_sources_ifdef(CONFIG_COMMON_LIBC_STRNLEN source/string/strnlen.c)
+
+# Prevent compiler from optimizing calloc into an infinite recursive call
+zephyr_library_cc_option(-fno-builtin-malloc)
```

Comment:
This needs to be properly abstracted for multiple toolchain support.

Reply:
@stephanosio I gave this a go, is it what you were wanting?
I am not familiar with some of these compilers, so I may have made a mistake. Synopsys seems to require an account to look at arcmwdt supported flags (https://www.synopsys.com/dw/doc.php/arc_tools/dw_arc_metaware/hw_id_core_versions.pdf)

Reply:
> I gave this a go, is it what you were wanting?

Thanks! That is exactly what I meant.

> I am not familiar with some of these compilers, so I may have made a mistake. Synopsys seems to require an account to look at arcmwdt supported flags

For now, only adding the support for the affected compiler(s) should be fine. The support for other proprietary compilers can be added later by the relevant maintainers (assuming they also require this flag). cc @abrodkin @evgeniy-paltsev